### PR TITLE
Fallback to aapt2 for string resources extraction

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This is the last PR, which completes transition from aapt to aapt2

### CODE REVIEW

File: test/unit/helper-specs.js [hunk 1]
```diff
@@ -322,4 +325,174 @@ describe('helpers', withMocks({fs}, function (mocks) {
...
+            (fr) "Horizontal"
+          resource 0x7f0c0217 string/linear_layout_8_top
+            () "Top"
+            (fr) "Top"
+      `;
+      const aapt2Strings = parseAapt2Strings(aapt2Output, '');
+      aapt2Strings.linear_layout_8_horizontal.should.eql('Horizontal');
+    });
+    it('should parse plurals received from aapt2 output', function () {
+      const aapt2Output = `
```
Comment (KazuCocoa) [inline test/unit/helper-specs.js]: 👍

File: test/unit/helper-specs.js [hunk 2]
```diff
@@ -322,4 +325,174 @@ describe('helpers', withMocks({fs}, function (mocks) {
...
+      aaptStrings.abc_action_bar_home_description.should.eql('Navigate "home"');
+      aaptStrings.calling__conversation_full__message.should.eql([
+        'Calls work in conversations with up to 1 person.',
+        'Calls work in conversations with up to %1$d people. "blabla"',
+      ]);
+    });
+  });
+
+  describe('parseAapt2Strings', function () {
+    it('should parse strings received from aapt2 output', function () {
```
Comment (KazuCocoa) [inline test/unit/helper-specs.js]: 👍

File: lib/helpers.js [hunk 3]
```diff
@@ -456,12 +457,230 @@ function parseManifest (manifestXml) {
...
+ * @param {string} defaultMarker The default config marker value
+ * @return {string} The formatted config marker
+ */
+async function formatConfigMarker (configsGetter, desiredMarker, defaultMarker) {
+  let configMarker = desiredMarker || defaultMarker;
+  if (configMarker.includes('-') && !configMarker.includes('-r')) {
+    configMarker = configMarker.replace('-', '-r');
+  }
+  if (configMarker.toLowerCase().startsWith('en')) {
+    // Assume the 'en' configuration is the default one
```
Comment (KazuCocoa) [inline lib/helpers.js]: 👍 
I think so. No vanilla `en` has like `en-rIN`.

File: test/unit/helper-specs.js [hunk 4]
```diff
@@ -322,4 +325,174 @@ describe('helpers', withMocks({fs}, function (mocks) {
...
+      Binary APK
+      Package name=io.appium.android.apis id=7f
+        type string id=0c entryCount=944
+          resource 0x7f0c0215 string/linear_layout_8_vertical
+            () "Vertical"
+            (fr) "Vertical"
+          resource 0x7f0c0216 string/linear_layout_8_horizontal
+            () "Horizontal"
+            (fr) "Horizontal"
+          resource 0x7f0c0217 string/linear_layout_8_top
```
Comment (KazuCocoa) [inline test/unit/helper-specs.js]: build tools 27 has a different format.
Haven't checked 26 etc, tho. 

```
kazu$ /Users/kazu/Library/Android/sdk/build-tools/27.0.3/aapt2 dump api.apk | grep "resource"
    spec resource 0x7f0e0087 io.appium.android.apis:string/admin_receiver_status_pw_changed
    spec resource 0x7f0e0088 io.appium.android.apis:string/admin_receiver_status_pw_failed
    spec resource 0x7f0e0089 io.appium.android.apis:string/admin_receiver_status_pw_succeeded
    spec resource 0x7f0e008a io.appium.android.apis:string/advanced_preferences
    spec resource 0x7f0e008b io.appium.android.apis:string/again
    spec resource 0x7f0e008c io.appium.android.apis:string/alarm_controller
    spec resource 0x7f0e008d io.appium.android.apis:string/alarm_service
    spec resource 0x7f0e008e io.appium.android.apis:string/alarm_service_finished
    spec resource 0x7f0e008f io.appium.android.apis:string/alarm_service_label
```

Comment (KazuCocoa) [reply test/unit/helper-specs.js]: So, https://github.com/appium/appium-adb/pull/467/files#diff-27a1c86133de4310373decfdad200a91R599 should be something like:

```diff
- if (trimmedLine.startsWith('resource')) {
+ if (/resource [\w]+/.test(trimmedLine)) {
```

?

Comment (mykola-mokhnach) [reply test/unit/helper-specs.js]: I've just checked aapt2 from build tools 27 and got the same output:

```
Nick:27.0.3 elf$ aapt2 dump ~/Desktop/ApiDemos-debug2.apk | grep resource
    resource 0x7f040000 anim/animator
    resource 0x7f040001 anim/animator_set
    resource 0x7f040002 anim/color_animator
    resource 0x7f040003 anim/cycle_7
    resource 0x7f040004 anim/fade
    resource 0x7f040005 anim/hold
```

what is the output of `aapt2 version` on your system? Mine shows 

```
$ aapt2 version
Android Asset Packaging Tool (aapt) 2:19
```

Comment (KazuCocoa) [reply test/unit/helper-specs.js]: mmm.
Mine was below

```
$ /Users/kazu/Library/Android/sdk/build-tools/27.0.3/aapt2 version
Android Asset Packaging Tool (aapt) 2:19
```

Comment (mykola-mokhnach) [reply test/unit/helper-specs.js]: These guys had the same output as me:
https://blog.csdn.net/iEearth/article/details/84632797

Perhaps, there is some hidden env flag, that influences that? I need to check aapt2 sources

Comment (mykola-mokhnach) [reply test/unit/helper-specs.js]: The recent aapt2 source should produce the same output: https://android.googlesource.com/platform/frameworks/base.git/+/master/tools/aapt2/Debug.cpp

Comment (KazuCocoa) [reply test/unit/helper-specs.js]: image

My env got them via AS environment.
From 28.0.0, I was able to get the same one.

```
    resource 0x7f0e012f string/dismiss
    resource 0x7f0e0130 string/display_options_custom_button
    resource 0x7f0e0131 string/display_options_menu_item
    resource 0x7f0e0132 string/dive
```

One way is raising an error to update user env at least 28.0.0 (Maybe)?

Comment (mykola-mokhnach) [reply test/unit/helper-specs.js]: we default to aapt, so should be fine for now

Comment (KazuCocoa) [reply test/unit/helper-specs.js]: Sounds reasonable

File: lib/helpers.js [hunk 5]
```diff
@@ -456,12 +457,234 @@ function parseManifest (manifestXml) {
...
+function parseAaptStrings (rawOutput, configMarker) {
+  const normalizeStringMatch = function (s) {
+    return s.replace(/"$/, '').replace(/^"/, '').replace(/\\"/g, '"');
+  };
+
+  const apkStrings = {};
+  let isInConfig = false;
+  let currentResourceId = null;
+  let isInPluralGroup = false;
+  const quotedStringPattern = /"[^"\\]*(?:\\.[^"\\]*)*"/;
```
Comment (imurchie) [inline lib/helpers.js]: The regular expression could use some explanation.

**Pull request merged**